lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -116,8 +116,15 @@ Flow firrtl::foldFlow(Value val, Flow accumulatedFlow) {
       // Registers, Wires, and behavioral memory ports are always Duplex.
       .Case<RegOp, RegResetOp, WireOp, MemoryPortOp>(
           [](auto) { return Flow::Duplex; })
-      .Case<InstanceOp>([&](auto) {
-        return val.getType().isa<FlipType>() ? swap() : accumulatedFlow;
+      .Case<InstanceOp>([&](auto inst) {
+        for (auto arg : llvm::enumerate(inst.getResults()))
+          if (arg.value() == val)
+            if (getModulePortDirection(inst.getReferencedModule(),
+                                       arg.index()) == Direction::Output)
```

Comment:
Is `getModulePortDirection` safe to use in a parallel context where modules are being operated on independently? (E.g., in LowerTypes?)